components/collection/drop/GenerativePreview.vue

```diff
@@ -85,40 +76,44 @@
 import { blake2AsHex, encodeAddress } from '@polkadot/util-crypto'
 import { NeoButton, NeoIcon } from '@kodadot1/brick'
 import { sanitizeIpfsUrl } from '@/utils/ipfs'
-import { DropItem } from '@/params/types'
-import type {
-  HolderOfCollectionProp,
-  MinimumFundsProp,
-  MintButtonProp,
-} from '@/components/collection/drop/types'
+import type { HolderOfCollectionProp } from '@/components/collection/drop/types'
 import { getRandomIntFromRange } from '../unlockable/utils'
 import { isValidSs58Format } from '@/utils/ss58Format'
 import useGenerativeIframeData from '@/composables/drop/useGenerativeIframeData'
+import { useDrop } from '@/components/drops/useDrops'
+import useGenerativeDropMint, {
+  useCollectionEntity,
+} from '@/composables/drop/useGenerativeDropMint'
 
-const props = defineProps<{
-  drop: DropItem
-  minted: number
-  collectionId: string
-  mintedCount: number
-  mintCountAvailable: boolean
-  maxCount: number
-  minimumFunds: MinimumFundsProp
-  isImageFetching: boolean
-  isWalletConnecting: boolean
-  isLoading: boolean
-  mintButton: MintButtonProp
+defineProps<{
   holderOfCollection?: HolderOfCollectionProp
 }>()
 
+const { drop } = useDrop()
+const dropStore = useDropStore()
+
+const { maxCount, mintedCount, mintCountAvailable } = useGenerativeDropMint()
+const { mintedAmountForCurrentUser } = useCollectionEntity()
+
+watch(
```

Comment:
safe to remove? or intended for debugging purposes?
can use `watchEffect` here?

Reply:
debug indeed
removed